documentation/cli/change_log.rst

```diff
@@ -12,6 +12,7 @@ since v0.12.0 | November XX, 2022
 * Fix ``flexmeasures db-ops dump`` and ``flexmeasures db-ops restore`` incorrectly reporting a success when `pg_dump` and `pg_restore` are not installed.
 * Add ``flexmeasures monitor last-seen``. 
 * Rename ``flexmeasures monitor tasks`` to ``flexmeasures monitor last-run``. 
+* Rename ``flexmeasures add schedule`` to ``flexmeasures add schedule-for-storage`` (in expectation of more scheduling commands, based on in-built flex models). 
```

Comment:
I don't understand why a breaking change is needed. In practice, wouldn't we infer the type of flexibility model from the asset type? And possibly let users override that in the future by passing a specific scheduler to be used (similar to the API), here as a CLI option.

Reply:
The problem is that we have to describe all these parameters on the CLI level. At least, we'd like to. On the API level, we refer to the docs and allow any dict as "flex_model" value. On the CLI level, we cannot nest parameters like this.

Reply:
I see. So we have to face this trade off: step away from conventional CLI documentation structure (1) or write a separate CLI command for each type of flexibility model (2).

(1) Similar to the API, we'd refer to the docs and allow any dict - as a JSON string - as "flex_model" value.
(2) How would this look/scale when we allow scheduling multiple flexible assets?

Reply:
Yes. I opted for (2), as passing JSON strings is not nice for developers, and we will not have a long list of supported flexibility models, I think.

I have no idea how multiple flex model support will look like right now.

Reply:
I think that, with multi-asset scheduling, too much information would need to be passed to stick with regular CLI options. But I agree that for less complex flex cases the regular CLI options are good to keep using. I have the feeling that there is a good trade-off possible, though, where we allow both approaches to nicely use the CLI documentation tooling without "cluttering" the `flexmeasures add` listing with scheduling commands.

What I have in mind is this:

```
flexmeasures add schedule storage --<regular CLI options for storages>
flexmeasures add schedule shiftable --<regular CLI options for shiftables>
flexmeasures add schedule --flex_model <json string (or file) for flex_model> --flex_context <json string (or file) for flex_context> --<regular CLI options such as start and duration>
```

This approach makes use of defining `schedule` as a subgroup, using:
```
@fm_add_data.group("schedule", invoke_without_command=True)
def create_schedule():
    if ctx.invoked_subcommand:
        ...
```
instead of:
```
@fm_add_data.command("schedule")
def create_schedule():
    ....
```

The benefits of this subgroup approach are that:
- it doesn't add a command to the listing for `flexmeasures add` for every new flexibility type, but only to the listing for `flexmeasures add schedule`.
- it still allows using CLI scheduling for individual flexible assets using conventional CLI documentation tooling.
- it also allows using CLI scheduling for more complex cases by passing flex_model and flex_context dictionaries as JSON strings.

I'd be happy to make a PR out of my tech spike, if you'd like (just enabling the subgroup implementation, not the JSON stuff).

Reply:
I like the sub-group idea. Tech spike very welcome.

The config-file option might be a path we need to take later. But not right now.